src/Context/Auth/index.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from 'react';
+import testUsers from './lib/users';
+import jwt_decode from "jwt-decode";
+import cookie from 'react-cookies';
+
+export const AuthContext = React.createContext();
+
+
+function AuthProvider({ children }){
+
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [user, setUser] = useState({});
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        let cookieToken = cookie.load('auth')
+        _validateToken(cookieToken);
+    }, []);
+
+    const _validateToken = (token) => {
+        try {
+            let validUser = jwt_decode(token);
+            if(validUser){
+                // save cookie
+                cookie.save('auth', token);
+                setUser(validUser);
+                setIsLoggedIn(true);
+            }
+        } catch(err){
+            setError(err);
+            console.log(err);
+        }
+    }
+
+    const login = (username, password) => {
+        let user = testUsers[username];
+        if(user && user.password === password){
+            try {
+                _validateToken(user.token);
+            } catch(e){
+                setError(e);
+                console.log(e);
+            }
+        }
+    }
+
+    const logout = () => {
+        setUser({});
+        setIsLoggedIn(false);
+    }
```

Comment:
When you log out, make sure you remove the cookie. Right now if you log out then refresh the page, it will automatically log you back in.